catalog/app/components/FileEditor/routes.ts

```diff
@@ -36,6 +39,20 @@ export function useEditFileInPackage(
   return editFileInPackage(urls, fileHandle, logicalKey, next)
 }
 
+export function useAddFileInPackage(packageHandle: PackageHandle, logicalKey: string) {
+  const { urls } = NamedRoutes.use<RouteMap>()
+  const { bucket, name } = packageHandle
```

Comment:
i think it's better to do this destructuring in the function signature, it will also fix memoization for the cases when a new object is passed with the same props (in current implementation memoization won't work for such cases)